tooling/internal/filename/parse_test.go

```diff
@@ -0,0 +1,39 @@
+package filename
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseFilename(t *testing.T) {
```

Comment:
Maybe add a semver prerelease test case?  We do get the odd ask to publish prereleases:

- https://github.com/gravitational/teleport/issues/7691
- https://github.com/gravitational/teleport-plugins/pull/444#issuecomment-1062346211

To be clear, I don't expect that we'll publish pre-releases in this swing -- I am thinking this may be an edge that someone would trip over in the future.